src/csharp/Microsoft.Spark/Sql/RowConstructor.cs

```diff
@@ -16,74 +18,116 @@ namespace Microsoft.Spark.Sql
     internal sealed class RowConstructor : IObjectConstructor
     {
         /// <summary>
-        /// Schema of the rows being received. Note that this is thread local variable
+        /// Cache the schema of the rows being received. Note that this is thread local variable
```

Comment:
Now you are storing `schemas`. Can you update the comment to explain why we need to have multiple schemas? It's for nested types?

Reply:
Added more comments.